### PR TITLE
Add runtimeconfig.json support to AppleAppBuilder, reenable affected tests

### LINKED ISSUE #50573: System.Text.Encoding.Tests fail on Android
Several tests from System.Text.Encoding.Tests fail on Android

- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"UTF7 Encoding Example\", index: 1, count: 2, expected: [84, 70])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xd800\\xd800\", index: 0, count: 2, expected: [43, 50, 65, 68, 89, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"abc\", index: 0, count: 0, expected: [])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"𐀀\", index: 0, count: 2, expected: [43, 50, 65, 68, 99, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xd800\", index: 0, count: 1, expected: [43, 50, 65, 65, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"𐏿\", index: 0, count: 2, expected: [43, 50, 65, 68, 102, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"�\", index: 0, count: 1, expected: [43, 47, 47, 48, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"abc\", index: 3, count: 0, expected: [])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"A\\t\\r\\n /z\", index: 0, count: 7, expected: [65, 9, 13, 10, 32, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xdfff\", index: 0, count: 1, expected: [43, 51, 47, 56, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"𐀀\", index: 0, count: 1, expected: [43, 50, 65, 65, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xffff\", index: 0, count: 1, expected: [43, 47, 47, 56, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"-\", index: 0, count: 1, expected: [45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"๙ב\", index: 0, count: 2, expected: [43, 68, 108, 107, 70, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xdc00\", index: 0, count: 1, expected: [43, 51, 65, 65, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"+-\", index: 0, count: 2, expected: [43, 45, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\t\\n\\rXYZabc123\", index: 0, count: 12, expected: [9, 10, 13, 88, 89, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"+\", index: 0, count: 1, expected: [43, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\", index: 0, count: 0, expected: [])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xdc00\\xd800\", index: 0, count: 2, expected: [43, 51, 65, 68, 89, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"Å\", index: 0, count: 1, expected: [43, 73, 83, 115, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xdc00\\xdc00\", index: 0, count: 2, expected: [43, 51, 65, 68, 99, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xfffe\", index: 0, count: 1, expected: [43, 47, 47, 52, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"ΠΣ\", index: 0, count: 2, expected: [43, 65, 54, 65, 68, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\f\", index: 0, count: 1, expected: [43, 65, 65, 119, 45])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"a𐀀b\", index: 0, count: 4, expected: [97, 43, 50, 65, 68, ...])
- [ ] System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"𐀀\", index: 1, count: 1, expected: [43, 51, 65, 65, 45])
- [ ] System.Text.Encodings.Tests.EncodingMiscTests.VerifyCodePageAttributes(codepage: 65000, name: \"utf-7\", bodyName: \"utf-7\", headerName: \"utf-7\", isBrowserDisplay: False, isBrowserSave: False, isMailNewsDisplay: True, isMailNewsSave: True, windowsCodePage: 1200)
- [ ] System.Text.Encodings.Tests.EncodingMiscTests.NormalizationTest(codepage: 65000, normalized: False, normalizedC: False, normalizedD: False, normalizedKC: False, normalizedKD: False)
- [ ] System.Text.Tests.EncodingGetEncodingTest.GetEncoding_WebName
- [ ] System.Text.Tests.EncodingGetEncodingTest.TestEncodingNameAndCopdepageNumber
- [ ] System.Text.Tests.EncodingGetEncodingTest.GetEncoding_EncodingName
- [ ] System.Text.Tests.UTF7EncodingTests.EqualsTest
- [ ] System.Text.Tests.UTF7EncodingTests.EncodingName
- [ ] System.Text.Tests.UTF7EncodingTests.CodePage
- [ ] System.Text.Tests.UTF7EncodingTests.Clone
- [ ] System.Text.Tests.UTF7EncodingTests.WebName
- [ ] System.Text.Tests.UTF7EncodingTests.IsSingleByte

Based on CI runs from https://github.com/dotnet/runtime/pull/50095

### LINKED ISSUE #51394: System.Text.Encoding.Tests Fails on iOS/tvOS
System.Text.Encoding.Tests
System.Text.Encodings.Tests.EncodingMiscTests.VerifyCodePageAttributes
System.Text.Encodings.Tests.EncodingMiscTests.NormalizationTest
System.Text.Tests.EncodingGetEncodingTest.GetEncoding_WebName
System.Text.Tests.EncodingGetEncodingTest.TestEncodingNameAndCopdepageNumber
System.Text.Tests.EncodingGetEncodingTest.GetEncoding_EncodingName
System.Text.Tests.UTF7EncodingEncode.Encode_Basic
System.Text.Tests.UTF7EncodingTests.EqualsTest
System.Text.Tests.UTF7EncodingTests.EncodingName
System.Text.Tests.UTF7EncodingTests.CodePage
System.Text.Tests.UTF7EncodingTests.Clone
System.Text.Tests.UTF7EncodingTests.WebName
System.Text.Tests.UTF7EncodingTests.IsSingleByte

```
System.Text.Encoding.Tests.dll   Failed: 38

Test collection for System.Text.Encodings.Tests.EncodingMiscTests
System.Text.Encodings.Tests.EncodingMiscTests.VerifyCodePageAttributes(codepage: 65000, name: \"utf-7\", bodyName: \"utf-7\", headerName: \"utf-7\", isBrowserDisplay: False, isBrowserSave: False, isMailNewsDisplay: True, isMailNewsSave: True, windowsCodePage: 1200)
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Encodings.Tests.EncodingMiscTests.NormalizationTest(codepage: 65000, normalized: False, normalizedC: False, normalizedD: False, normalizedKC: False, normalizedKD: False)
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.

Test collection for System.Text.Tests.EncodingGetEncodingTest
System.Text.Tests.EncodingGetEncodingTest.GetEncoding_WebName
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.EncodingGetEncodingTest.TestEncodingNameAndCopdepageNumber
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.EncodingGetEncodingTest.GetEncoding_EncodingName
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.

Test collection for System.Text.Tests.UTF7EncodingEncode
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"UTF7 Encoding Example\", index: 1, count: 2, expected: [84, 70])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xd800\\xd800\", index: 0, count: 2, expected: [43, 50, 65, 68, 89, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"abc\", index: 0, count: 0, expected: [])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"𐀀\", index: 0, count: 2, expected: [43, 50, 65, 68, 99, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xd800\", index: 0, count: 1, expected: [43, 50, 65, 65, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"𐏿\", index: 0, count: 2, expected: [43, 50, 65, 68, 102, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"�\", index: 0, count: 1, expected: [43, 47, 47, 48, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"abc\", index: 3, count: 0, expected: [])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"A\\t\\r\\n /z\", index: 0, count: 7, expected: [65, 9, 13, 10, 32, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xdfff\", index: 0, count: 1, expected: [43, 51, 47, 56, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"𐀀\", index: 0, count: 1, expected: [43, 50, 65, 65, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xffff\", index: 0, count: 1, expected: [43, 47, 47, 56, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"-\", index: 0, count: 1, expected: [45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"๙ב\", index: 0, count: 2, expected: [43, 68, 108, 107, 70, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xdc00\", index: 0, count: 1, expected: [43, 51, 65, 65, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"+-\", index: 0, count: 2, expected: [43, 45, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\t\\n\\rXYZabc123\", index: 0, count: 12, expected: [9, 10, 13, 88, 89, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"+\", index: 0, count: 1, expected: [43, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\", index: 0, count: 0, expected: [])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xdc00\\xd800\", index: 0, count: 2, expected: [43, 51, 65, 68, 89, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"Å\", index: 0, count: 1, expected: [43, 73, 83, 115, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xdc00\\xdc00\", index: 0, count: 2, expected: [43, 51, 65, 68, 99, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\xfffe\", index: 0, count: 1, expected: [43, 47, 47, 52, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"ΠΣ\", index: 0, count: 2, expected: [43, 65, 54, 65, 68, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"\\f\", index: 0, count: 1, expected: [43, 65, 65, 119, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"a𐀀b\", index: 0, count: 4, expected: [97, 43, 50, 65, 68, ...])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingEncode.Encode_Basic(source: \"𐀀\", index: 1, count: 1, expected: [43, 51, 65, 65, 45])
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.

Test collection for System.Text.Tests.UTF7EncodingTests
System.Text.Tests.UTF7EncodingTests.EqualsTest
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingTests.EncodingName
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingTests.CodePage
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingTests.Clone
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingTests.WebName
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.
System.Text.Tests.UTF7EncodingTests.IsSingleByte
    System.NotSupportedException : Support for UTF-7 is disabled. See https://aka.ms/dotnet-warnings/SYSLIB0001 for more information.

### PR DESCRIPTION
**Pull request by filipnavara**
Fixes #50573
Fixes #51394

It's quite unexpected to me that overriding methods through `runtimeconfig.template.json` doesn't work on WASM/iOS/Android. Some of the properties can be specified with both MSBuild properties and the JSON config file. The MSBuild properties take precedence over the user specified JSON file. Since the new SDKs specify defaults for the MSBuild properties the defaults always override the JSON config mechanism.

### CODE REVIEW

**Comment by steveisok** [comment]
The android failures are a known issue https://github.com/dotnet/xharness/issues/597

The wasm failure is also know https://github.com/dotnet/runtime/issues/50965

**Review by steveisok — APPROVED** [review]
LGTM.  The linker tests are impacted somehow with the path to the runtimeconfig task.

**Comment by filipnavara** [comment]
I'll look at the linker tests... or just condition the task/target to only run on mobile.

File: src/tasks/AndroidAppBuilder/Templates/monodroid.c [hunk 1]
```diff
@@ -239,6 +239,8 @@ mono_droid_runtime_init (const char* executable, int managed_argc, char* managed
         arg->kind = 0;
         arg->runtimeconfig.name.path = file_path;
         monovm_runtimeconfig_initialize (arg, cleanup_runtime_config, file_path);
+    } else {
+        free (file_path);
```
Comment (CoffeeFlux) [inline src/tasks/AndroidAppBuilder/Templates/monodroid.c:243]: Good catch, thanks.

**Review by CoffeeFlux — APPROVED** [review]
Native bits LGTM, thanks!

**Review by fanyang-mono — COMMENTED** [review]
The change looks good to me. Thanks for adding  runtimeconfig.json support to AppleAppBuilder.

**Pull request merged**